### PR TITLE
Rewriting RTIR to work without RT package

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
In Progress
## Related Issues
fixes: https://github.com/demisto/etc/issues/15067

## Does it break backward compatibility?
   - No

## Must have
- [ ] Code Review

### CODE REVIEW

**Comment by Shellyber** [comment]
Working on the Fetch incidents command

File: Integrations/integration-RTIR.yml [hunk 1]
```diff
@@ -59,11 +59,10 @@ script:
     import requests
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: remove imports which are not used

File: Integrations/integration-RTIR.yml [hunk 2]
```diff
@@ -165,22 +172,41 @@ script:
                 kwargs[key] = value
 
         attachments = demisto.args().get('attachment')
-        files = []
         if attachments:
+            files_data = {}
             if isinstance(attachments, list): # Given as list
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: use `argToList`

File: Integrations/integration-RTIR.yml [hunk 3]
```diff
@@ -585,16 +684,54 @@ script:
...
+
     ''' EXECUTION CODE '''
 
-    tracker = rt.Rt(BASE_URL, USERNAME, PASSWORD, verify_cert=USE_SSL)
-    login = tracker.login()
-    if login is False:
-        return_error('Failed to login')
 
-    LOG('command is %s' % (demisto.command(), ))
+    def http_request(method, suffix_url, params=None, data=None, files=None):
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: add the credentials param to here instead of passing them from each function

File: Integrations/integration-RTIR.yml [hunk 4]
```diff
@@ -585,16 +684,54 @@ script:
...
+            return_error('Error in API call with status code {}\n{}'.format(response.status_code, message))
+
+        return response
+
+
+    def parse_error_response(response):
+        try:
+            res = response.json()
+            msg = res.get('message')
+            if res.get('details') is not None and res.get('details')[0].get('message') is not None:
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: no need to check `is not None`, but `res.get('details)` is enough, same for the second part of the condition

File: Integrations/integration-RTIR.yml [hunk 5]
```diff
@@ -585,16 +684,54 @@ script:
...
+                msg = msg + "\n"+json.dumps(res.get('details')[0])
+        except:
+            return response.text
+        return msg
+
+
+    def login():
+
+        suffix_url = ''
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: this is not used, remove it

File: Integrations/integration-RTIR.yml [hunk 6]
```diff
@@ -585,16 +684,54 @@ script:
             demisto.setLastRun({'ticket_id': max_ticket_id})
         demisto.incidents(incidents)
 
+
     ''' EXECUTION CODE '''
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: the `http_request`, `login` and `logout` should not be in the execution section, but in helper functions section

File: Integrations/integration-RTIR.yml [hunk 7]
```diff
@@ -110,50 +112,55 @@ script:
...
 
         priority = demisto.args().get('priority')
         if priority:
-            kwargs['Priority'] = int(priority)
+            data += "Priority: {}".format(priority)
 
         initial_priority = demisto.args().get('initial-priority')
         if initial_priority:
-            kwargs['InitialPriority'] = int(initial_priority)
+            data += "Initiatl-priority: {}".format(initial_priority)
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: there seems to be a typo here..

File: Integrations/integration-RTIR.yml [hunk 8]
```diff
@@ -110,50 +112,55 @@ script:
...
 
         cc = demisto.args().get('cc', '')
         if cc:
-            kwargs['Cc'] = cc
+            data += "Cc: {}".format(cc)
 
         admin_cc = demisto.args().get('admin-cc', '')
         if admin_cc:
-            kwargs['AdminCc'] = admin_cc
+            data += "Admin-Cc: {}".format(admin_cc)
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: are you sure it's not `AdminCc` ?

File: Integrations/integration-RTIR.yml [hunk 9]
```diff
@@ -165,22 +172,41 @@ script:
...
+            files_data.update({'content': (None, data)})
+            ticket_id = http_request('POST', suffix_url, params={'user': USERNAME, 'pass': PASSWORD}, files=files_data)
+        else:
+            ticket_id = http_request('POST', suffix_url, data=encoded, params={'user': USERNAME, 'pass': PASSWORD})
+        ticket_id = re.findall('\d+', ticket_id.content)[-1]
         if ticket_id == -1:
             return_error('Ticket creation failed')
 
-        kwargs['ID'] = ticket_id
+        data += "ID: {}".format(ticket_id)
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: what is this for ?

File: Integrations/integration-RTIR.yml [hunk 10]
```diff
@@ -165,22 +172,41 @@ script:
...
+            'Subject': subject,
+            'Creator': requestor,
+            'InitialPriority': initial_priority,
+            'Priority': priority,
+            'FinalPriority': final_priority,
+            'Owner': owner
+        })
         ec= {
-            'RTIR.Ticket(val.ID && val.ID == obj.ID)': kwargs
+            'RTIR.Ticket(val.ID && val.ID == obj.ID)': ticket_context
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: `==` -> `===`

File: Integrations/integration-RTIR.yml [hunk 11]
```diff
@@ -192,6 +218,14 @@ script:
             'EntryContext': ec
         })
 
+    def ticket_request(ticket_id):
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: improve function name, like `search_ticket` and the other one to be `search_ticket_request`

File: Integrations/integration-RTIR.yml [hunk 12]
```diff
@@ -240,55 +274,60 @@ script:
...
-                    if raw_ticket[key]:
-                        custom_field_regex = re.findall(CURLY_BRACKETS_REGEX, key)[0].replace(' ', '') # Regex and removing white spaces
-                        ticket[custom_field_regex] = raw_ticket[key]
-                        headers.append(custom_field_regex)
-            tickets.append(ticket)
-        if tickets:
+        if queue:
+            raw_query += 'Queue={}{}{}+AND+'.format(apostrophe, queue, apostrophe)
+        suffix_url = 'search/ticket'
+        headers = ['id', 'Subject', 'Status', 'Priority', 'Created', 'Queue', 'Creator', 'Owner', 'InitialPriority', 'FinalPriority']
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: initialize headers before the entry
should be `ID` and not `id`

File: Integrations/integration-RTIR.yml [hunk 13]
```diff
@@ -240,55 +274,60 @@ script:
...
+            search_ticket = search_ticket[2:]
+            current_ticket_search = {}
+            for entity in search_ticket:
+                if ': ' in entity:
+                    header, content = entity.split(': ', 1)
+                    if header in {'id', 'Subject', 'Status', 'Priority', 'Created', 'Queue', 'Creator', 'Owner', 'InitialPriority', 'FinalPriority'}:
+                        current_ticket_search[header] = content
+            if current_ticket_search:
+                search_context.append(current_ticket_search)
+        # for key in raw_tickets: # Adding ticket custom fields to outputs
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: remove comments

File: Integrations/integration-RTIR.yml [hunk 14]
```diff
@@ -240,55 +274,60 @@ script:
...
             }
             title = 'RTIR ticket search results'
             demisto.results({
                 'Type': entryTypes['note'],
-                'Contents': raw_tickets,
+                'Contents': search_context,
                 'ContentsFormat': formats['json'],
                 'ReadableContentsFormat': formats['markdown'],
-                'HumanReadable': tableToMarkdown(title, tickets, headers),
+                'HumanReadable': tableToMarkdown(title, search_context, headers),
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: add `removeNull=True` to all `tableToMarkdown`

File: Integrations/integration-RTIR.yml [hunk 15]
```diff
@@ -240,55 +274,60 @@ script:
...
                 'EntryContext': ec
             })
         else:
             demisto.results('No results found.')
 
     def close_ticket():
         ticket_id = demisto.args().get('ticket-id')
+        suffix_url = 'ticket/{}/edit'.format(ticket_id)
+        content = ''
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: this is not necessary, remove it

File: Integrations/integration-RTIR.yml [hunk 16]
```diff
@@ -240,55 +274,60 @@ script:
...
+        content += '\nStatus: resolved'
         try:
-            closed_ticket = tracker.edit_ticket(ticket_id=ticket_id, Status='resolved')
-            if closed_ticket is True:
+            encoded = "content=" + urllib.quote_plus(content)
+            closed_ticket = http_request('POST', suffix_url, params={'user': USERNAME, 'pass': PASSWORD}, data=encoded)
+            if closed_ticket.status_code in {200}:
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: why is this needed ? it is checked in `http_request`

File: Integrations/integration-RTIR.yml [hunk 17]
```diff
@@ -240,55 +274,60 @@ script:
...
             })
         else:
             demisto.results('No results found.')
 
     def close_ticket():
         ticket_id = demisto.args().get('ticket-id')
+        suffix_url = 'ticket/{}/edit'.format(ticket_id)
+        content = ''
+        content += '\nStatus: resolved'
         try:
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: why is this needed ?

File: Integrations/integration-RTIR.yml [hunk 18]
```diff
@@ -360,14 +405,24 @@ script:
...
+        suffix_url = 'ticket/{}/edit'.format(ticket_id)
         if arguments_given:
             try:
-                edited_ticket = tracker.edit_ticket(ticket_id=ticket_id, **kwargs)
-                if edited_ticket is True:
-                    ec= {
-                        'RTIR.Ticket(val.ID && val.ID == obj.ID)': ticket
+                encoded = "content=" + urllib.quote_plus(content)
+                edited_ticket = http_request('POST', suffix_url, params={'user': USERNAME, 'pass': PASSWORD}, data=encoded)
+                if edited_ticket.status_code in {200}:
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: same here

File: Integrations/integration-RTIR.yml [hunk 19]
```diff
@@ -387,25 +442,42 @@ script:
 
 
     def get_ticket_attachments(ticket_id):
-        raw_attachments = tracker.get_attachments(ticket_id)
+        suffix_url = 'ticket/{}/attachments'.format(ticket_id)
+        raw_attachments = http_request('GET', suffix_url, params={'user': USERNAME, 'pass': PASSWORD}).content
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: if you always need `.content` then just return it from `http_request`

File: Integrations/integration-RTIR.yml [hunk 20]
```diff
@@ -426,138 +498,165 @@ script:
...
-                current_history['NewValue'] = new_value
-                if not 'NewValue' in headers:
-                    headers.append('NewValue')
-            output_history.append(current_history)
-        return output_history, raw_history, headers
+
+        suffix_url = 'ticket/{}/history'.format(ticket_id)
+        raw_history = http_request('GET', suffix_url, params={'user': USERNAME, 'pass': PASSWORD})
+        history_context = []
+        headers = ['id','Created', 'Creator', 'Description']
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: initialize `headers` before entry
this should be `id`, add white space after comma

File: Integrations/integration-RTIR.yml [hunk 21]
```diff
@@ -426,138 +498,165 @@ script:
...
+        data = raw_history.text.split("\n")
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: `"` -> `'`

File: Integrations/integration-RTIR.yml [hunk 22]
```diff
@@ -426,138 +498,165 @@ script:
...
-            'InitialPriority': int(raw_ticket.get('InitialPriority')),
-            'FinalPriority': int(raw_ticket.get('FinalPriority')),
-            'Queue': raw_ticket.get('Queue'),
-            'Owner': raw_ticket.get('Owner')
-        }
-        due = raw_ticket.get('Due')
-        if due != 'Not set':
-            ticket['Due'] = due
-            headers.append('Due')
+        headers = ['id', 'Subject', 'Status', 'Priority', 'Created', 'Queue', 'Creator', 'Owner', 'InitialPriority', 'FinalPriority']
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: same here

File: Integrations/integration-RTIR.yml [hunk 23]
```diff
@@ -426,138 +498,165 @@ script:
...
         if raw_links:
             links = []
             for raw_link in raw_links:
-                link_id = int(raw_link.rsplit('/', 1)[-1])
+                link_id = raw_link.rsplit('/', 3)[-2]
                 links.append({
                     'ID': link_id
                 })
-            ticket['LinkedTo'] = links
+            ticketProperties['LinkedTo'] = links
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: use PEP8 convention

File: Integrations/integration-RTIR.yml [hunk 24]
```diff
@@ -426,138 +498,165 @@ script:
...
+                files_data['attachment_{:d}'.format(i + 1)] = (file_name ,open(file['path'], 'rb'))
+                content += 'Attachment: {}'.format(file_name)
+
         try:
-            added_comment = tracker.comment(ticket_id=ticket_id, text=text, files=files)
-            if added_comment is True:
+            encoded = "content=" + urllib.quote_plus(content)
+            suffix_url = 'ticket/{}/comment'.format(ticket_id)
+            added_comment = http_request('POST', suffix_url, params={'user': USERNAME, 'pass': PASSWORD}, data=encoded)
+            if "200" in added_comment.content:
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: `"` -> `'`

File: Integrations/integration-RTIR.yml [hunk 25]
```diff
@@ -426,138 +498,165 @@ script:
...
+
     def add_reply():
         ticket_id = demisto.args().get('ticket-id')
+        content = 'Action: comment\n'
         text = demisto.args().get('text')
+        if text:
+            content += '\nText: ' + text.encode('utf-8')
         cc = demisto.args().get('cc')
+        if cc:
+            content += '\ncc: ' + cc
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: shouldnt this be `Cc` ?

File: Integrations/integration-RTIR.yml [hunk 26]
```diff
@@ -570,7 +669,7 @@ script:
                 status_query += 'Status={}{}{}+OR+'.format(apostrophe, status, apostrophe)
             raw_query += status_query
             raw_query += ')'
```
Comment (Itay4) [inline Integrations/integration-RTIR.yml]: unify the two strings concatenate
